File: .NET/VS2010TrainingKit/Labs/IntroToWF/Source/Ex9-ActivityDesigner/End/VB/HelloWorkflow.Activities.Tests/PreOnlyWorkflow.vb
' ----------------------------------------------------------------------------------
' Microsoft Developer & Platform Evangelism
' 
' Copyright (c) Microsoft Corporation. All rights reserved.
' 
' THIS CODE AND INFORMATION ARE PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
' EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED WARRANTIES 
' OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
' ----------------------------------------------------------------------------------
' The example companies, organizations, products, domain names,
' e-mail addresses, logos, people, places, and events depicted
' herein are fictitious.  No association with any real company,
' organization, product, domain name, email address, logo, person,
' places, or events is intended or should be inferred.
' ----------------------------------------------------------------------------------

Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Activities
Imports HelloWorkflow.Activities
Imports System.Activities.Statements

Namespace HelloWorkflow.Activities.Tests

    Public NotInheritable Class PreOnlyWorkflow
        Inherits Activity(Of String)
        Public Const PreMsg As String = "Pre Activity Called"

        Public Sub New()
            MyBase.Implementation = New Func(Of Activity)(AddressOf CreateBody)
        End Sub

        Private Function CreateBody() As Activity
            Return New PrePostSequence()
        End Function
    End Class
End Namespace

File: .NET/VS2010TrainingKit/Labs/IntroToWF/Source/Ex10-HostedDesigner/Begin/C#/HelloWorkflow.Activities.Tests/PreOnlyWorkflow.cs
// ----------------------------------------------------------------------------------
// Microsoft Developer & Platform Evangelism
// 
// Copyright (c) Microsoft Corporation. All rights reserved.
// 
// THIS CODE AND INFORMATION ARE PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
// EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED WARRANTIES 
// OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
// ----------------------------------------------------------------------------------
// The example companies, organizations, products, domain names,
// e-mail addresses, logos, people, places, and events depicted
// herein are fictitious.  No association with any real company,
// organization, product, domain name, email address, logo, person,
// places, or events is intended or should be inferred.
// ----------------------------------------------------------------------------------

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Activities;
using HelloWorkflow.Activities;
using System.Activities.Statements;

namespace HelloWorkflow.Activities.Tests
{

    public sealed class PreOnlyWorkflow : Activity<string>
    {
        public const string PreMsg = "Pre Activity Called";

        public PreOnlyWorkflow()
        {
            base.Implementation = new Func<Activity>(CreateBody);
        }

        Activity CreateBody()
        {
            return new PrePostSequence()
            {
                Pre = new Assign() 
                { 
                    To = new OutArgument<string>(env => this.Result.Get(env)), 
                    Value = new InArgument<string>(PreMsg) 
                }
            };
        }
    }
}
